### PR TITLE
Simplify AmpStoryHint (fixes repaint issue)

### PR DESCRIPTION
**Pull request by alanorozco**
- Simplifies hide/show/timeout lifecycle.
- Some code style changes/refactor.

### CODE REVIEW

File: extensions/amp-story/0.1/amp-story-hint.css [hunk 1]
```diff
@@ -17,6 +17,17 @@
 .i-amphtml-story-hint-container {
   transition-property: opacity !important;
   transition-duration: 200ms !important;
+  contain: strict !important;
```
Comment (choumx) [inline extensions/amp-story/0.1/amp-story-hint.css]: X-browser support is inconsistent: https://caniuse.com/#feat=css-containment

File: extensions/amp-story/0.1/amp-story-hint.js [hunk 2]
```diff
@@ -161,18 +153,29 @@ export class AmpStoryHint {
...
+      this.timer_.cancel(this.hintTimeout_);
+      this.hintTimeout_ = null;
+    }
   }
 
   /** @private */
-  resetContainerStyle_() {
-    this.hintContainer_.classList.remove(NAVIGATION_OVERLAY_CLASS);
-    this.hintContainer_.classList.remove(FIRST_PAGE_OVERLAY_CLASS);
+  fadeOut_() {
```
Comment (choumx) [inline extensions/amp-story/0.1/amp-story-hint.js]: This hides the hints immediately right? How about just `hide_()`?

Comment (alanorozco) [reply extensions/amp-story/0.1/amp-story-hint.js]: Sure, `hideInternal_` to discern from the public method.

File: extensions/amp-story/0.1/amp-story-hint.css (hunk 1 above)
Comment (choumx) [reply extensions/amp-story/0.1/amp-story-hint.css]: Why not? Is this just nice-to-have?

Comment (alanorozco) [reply extensions/amp-story/0.1/amp-story-hint.css]: Correct

**Pull request merged**